client/rpc/src/lib.rs

```diff
@@ -0,0 +1,150 @@
+use jsonrpsee::core::async_trait;
+use jsonrpsee::core::RpcResult;
+use jsonrpsee::proc_macros::rpc;
+use jsonrpsee::types::error::ErrorObjectOwned as JsonRpseeError;
+use jsonrpsee::types::error::INTERNAL_ERROR_CODE;
+use jsonrpsee::types::error::INTERNAL_ERROR_MSG;
+use jsonrpsee::types::ErrorObjectOwned;
+
+use shc_common::types::FileMetadata;
+use shc_common::types::FILE_CHUNK_SIZE;
+use sp_runtime::AccountId32;
+use sp_trie::TrieLayout;
+
+use file_manager::traits::FileDataTrie;
+use file_manager::traits::FileStorage;
+
+use log::debug;
+use log::error;
+
+use std::fmt::Debug;
+use std::fs::File;
+use std::io::Read;
+use std::marker::PhantomData;
+use std::path::PathBuf;
+use std::sync::Arc;
+
+use tokio::sync::RwLock;
+
+const LOG_TARGET: &str = "file-storage-rpc";
+
+/// Provides an interface with the desired RPC method.
+/// Used by the `rpc` macro from `jsonrpsee`
+/// to generate the trait that is actually going to be implemented.
+#[rpc(server, namespace = "filestorage")]
+#[async_trait]
+pub trait FileStorageApi {
+    #[method(name = "loadFileInStorage")]
+    async fn load_file_in_storage(
+        &self,
+        file_path: String,
+        location: String,
+        owner: AccountId32,
+    ) -> RpcResult<FileMetadata>;
+}
+
+/// Stores the required objects to be used in our RPC method.
+pub struct FileStorageRpc<FL, T> {
+    file_storage: Arc<RwLock<FL>>,
+    _marker: PhantomData<T>,
+}
+
+impl<FL, T> FileStorageRpc<FL, T> {
+    pub fn new(file_storage: Arc<RwLock<FL>>) -> Self {
+        Self {
+            file_storage,
+            _marker: Default::default(),
+        }
+    }
+}
+
+/// Interface generated by the `rpc` macro from our `FileStorageApi` trait.
+// TODO: Currently the UserSendsFile task will react to all runtime events triggered by
+// file uploads, even if the file is not in its storage. So we need a way to inform the task
+// to only react to its file.
+#[async_trait]
+impl<FL, T> FileStorageApiServer for FileStorageRpc<FL, T>
+where
+    FL: Send + Sync + FileStorage<T>,
+    T: Send + Sync + TrieLayout + 'static,
+{
+    async fn load_file_in_storage(
+        &self,
+        file_path: String,
+        location: String,
+        owner: AccountId32,
+    ) -> RpcResult<FileMetadata> {
+        // Open file in the local file system.
+        let mut file = File::open(PathBuf::from(file_path.clone())).map_err(into_rpc_error)?;
```

Comment:
I believe we should store the files with the names being the `file_key`. This would avoid any possible collisions with other uploaded files. It would be more ergonomic to use `file_key` across all of our query methods.

Reply:
Once we incorporate MSPs and the concept of buckets, a `file_path` should be constructed from the `bucket-id/arbitrary/path`. At that point, if a file path is the same, as far as StorageHub is concerned, the user is replacing a file.

Reply:
For a BSP however, it could be useful to have a mapping from `file_key` to `file_location`. But that is out of scope for this "user side" RPC method.